common/persistence/execution_manager.go

```diff
@@ -426,19 +427,19 @@ func (m *executionManagerImpl) SerializeWorkflowMutation(
 
 	var err error
 
-	transferTasks, err := m.serializer.SerializeTransferTasks(input.TransferTasks)
+	transferTasks, err := m.serializer.SerializeTransferTasks(input.Tasks[tasks.CategoryTransfer])
```

Comment:
do `.Tasks[tasks.CategoryTransfer]` inside SerializeTransferTasks?

Reply:
That will require changes to the serializer interface and kinda not relevant to task generation.
These methods on serializer (SerializeXXXTasks) will be replaced with more general SerializeTasks methods soon, so to me it seems unnecessary to change them just for this PR.
But let me know if you have strong opinion on it.